src/plots/cartesian/graph_interact.js

```diff
@@ -59,7 +59,7 @@ exports.initInteractions = function initInteractions(gd) {
                 // This is on `gd._fullLayout`, *not* fullLayout because the reference
                 // changes by the time this is called again.
                 gd._fullLayout._rehover = function() {
-                    if(gd._fullLayout._hoversubplot === subplot) {
+                    if((gd._fullLayout._hoversubplot === subplot) && gd._fullLayout._plots[subplot]) {
```

Comment:
What about putting this condition test inside `Fx.hover`?

Reply:
Why would that be better?

Reply:
Just in case that the function is called from different places.
Otherwise; this performs better.

Reply:
You're making a good point here, consider:

```js
        /*var data = [
            {y: [1], hoverinfo: 'y'},
            {y: [2], hoverinfo: 'y', xaxis: 'x2', yaxis: 'y2'}
        ];*/

        var data2 = [
            {y: [1], hoverinfo: 'y'}
        ];

        var layout = {
            xaxis: {domain: [0, 0.5]},
            xaxis2: {anchor: 'y2', domain: [0.5, 1]},
            yaxis2: {anchor: 'x2'},
            width: 400,
            height: 200,
            margin: {l: 0, t: 0, r: 0, b: 0},
            showlegend: false
        };
Plotly.newPlot(gd, data2, layout)

// which is part of our semi-official API
Plots.Fx.hover(gd, {}, 'x2y2')
```

fails with the same `Cannot read _subplot of undefined ` error.

---

So, I'll keep the check and `graph_interact.js` (to not have to call `Fx.hover` for nothing) AND  handle the empty subplot case in `Fx.hover` (+ a test).

Thanks for the headsup @archmoj !

Reply:
done in https://github.com/plotly/plotly.js/pull/4269/commits/89e52f71eb663bbaa0e045757c78c38454991569